File: two_sum.cpp
#include<vector>
use namesp std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for (int i = 0; i < nums.size(); i++) {
            int req = target - nums[i];
            if (mp.find(req) != mp.end()) {
                return {i, mp[req]};
            }
            mp[nums[i]] = i;
        }

        return {}; // No solution found
    }
};

File: twosum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp= new HashMap<>();
        int n=nums.length;
        for (int i=0;i<n;i++){
            int req=target - nums[i];
            if(mp.containsKey(req)){
                return new int[]{i,mp.get(req)};
            }
            mp.put(nums[i],i);

        }
        return null;
    }
}